### PR TITLE
[Concept Entry] Rust Data Types #3373

### PR DESCRIPTION
**Pull request by moha230**
### Description

- added concept entry rust data-types
- added a folder data-types 
- added a file data-types.md
- Closes issue  #3373 

### Type of Change

- Adding concept entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR

### CODE REVIEW

**Comment by moha230** [comment]
Hello @ishg-153  and @dakshdeepHERE its done

File: content/rust/concepts/data-types/data-types.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  - 'Conditionals'
+  - 'Control Flow'
+CatalogContent:
+  - 'rust-for-programmers'
+  - 'paths/computer-science'
+---
+
+**Data types** define the nature of data a variable can store. Rust employs static typing, ensuring that variable types are known at compile time, aiding in detecting potential errors before runtime. These data types are broadly categorized into two groups: primitive types and compound types.
+
+## Primitive types || scala types
```
Comment (dakshdeepHERE) [inline content/rust/concepts/data-types/data-types.md:17]:
```suggestion
## Primitive and Scala types
```

File: content/rust/concepts/data-types/data-types.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+```rust
+fn main() {
+// Floating-Point Type
+    let float_num: f64 = 3.14;
+    }
+```
+
+### Boolean
+
+A boolean is a data type representing one of two values: `true` or `false`. Booleans have a size of one byte and they are specified using the `bool` keyword with explicit type annotation:
```
Comment (dakshdeepHERE) [inline content/rust/concepts/data-types/data-types.md:45]: give a funtion returning bool value example

File: content/rust/concepts/data-types/data-types.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    println!("Boolean: {}", is_rust_cool);
+    println!("Character: {}", unicode_char);
+}
+```
+
+## Compound Types
+
+Compound types represent multiple values and can be grouped.
+
+## Tuples
```
Comment (dakshdeepHERE) [inline content/rust/concepts/data-types/data-types.md:77]:
```suggestion
### Tuples
```

File: content/rust/concepts/data-types/data-types.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    let person: (&str, usize, bool) = ("Alice", 30, true);
+
+    // Accessing tuple elements
+    let name = person.0;
+    let age = person.1;
+    let is_adult = person.2;
+    }
+```
+
+## Arrays
```
Comment (dakshdeepHERE) [inline content/rust/concepts/data-types/data-types.md:93]:
```suggestion
### Arrays
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
For now I have few comments for you to work on

**Comment by dakshdeepHERE** [comment]
Also really well done on fixing the commit issue @moha230

**Comment by ishg-153** [comment]
@dakshdeepHERE let me know when this is ready for a second review.

**Comment by moha230** [comment]
its done @dakshdeepHERE

**Comment by moha230** [comment]
hello @ishg-153 its done

**Comment by moha230** [comment]
Hello @dakshdeepHERE i have done the changes is it possible to inform @ishg-153 thank you in advance.

File: content/rust/concepts/data-types/data-types.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  - 'Conditionals'
+  - 'Control Flow'
+CatalogContent:
+  - 'rust-for-programmers'
+  - 'paths/computer-science'
+---
+
+**Data types** define the nature of data a variable can store. Rust employs static typing, ensuring that variable types are known at compile time, aiding in detecting potential errors before runtime. These data types are broadly categorized into two groups: primitive types and compound types.
+
+## Primitive and Scala types
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:17]:
```suggestion
## Primitive and Scalar types
```

File: content/rust/concepts/data-types/data-types.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+Scala types represent a single value.
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:19]:
```suggestion
Scalar types represent a single value.
```

File: content/rust/concepts/data-types/data-types.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+### Integer
+
+An integer in Rust is a number without a fractional component. The signed integer starts with an `i` followed by the bit number and the unsigned integer starts with `u` followed by the bit number. The built-in integer types in Rust include 8bit, 16bit, 32bit, 64bit, and 128bit:
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:23]:
```suggestion
An integer in Rust is a number without a fractional component. The signed integer starts with an `i` followed by the bit number and the unsigned integer starts with `u` followed by the bit number. The built-in integer types in Rust include `8bit`, `16bit`, `32bit`, `64bit`, and `128bit`:
```

File: content/rust/concepts/data-types/data-types.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+```rust
+fn main() {
+// Integer Type
+    let integer_num: i32 = 42;
+    }
+```
+
+### Floating-Point
+
+A floating-point represents numbers with decimal points. The built-in floating-point types are `f32` and `f64` which are 32 bits and 64 bits in size respectively. The default size is `f64` and all floating points are signed:
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:34]:
```suggestion
A floating point represents numbers with decimal points. The built-in floating-point types are `f32` and `f64` which are 32 bits and 64 bits in size respectively. The default size is `f64` and all floating points are signed:
```

File: content/rust/concepts/data-types/data-types.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+let unicode_char: char = 'A';
+
+    // Printing the values
+    println!("Integer: {}", integer_num);
+    println!("Floating-Point: {}", float_num);
+    println!("Boolean: {}", is_rust_cool);
+    println!("Character: {}", unicode_char);
+}
+```
+
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:89]: Can you add the output of the above code here?

File: content/rust/concepts/data-types/data-types.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+## Compound Types
+
+Compound types represent multiple values and can be grouped.
+
+### Tuples
+
+A tuple is a collection of values of different types. Tuples are constructed using parentheses `()` and each has a value with type signature. Tuples are used by functions to return multiple values as tuples can hold any number of values:
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:96]:
```suggestion
A tuple is a collection of values of different types. Tuples are constructed using parentheses `()` and each has a value with the type signature. Tuples are used by functions to return multiple values as tuples can hold any number of values:
```

File: content/rust/concepts/data-types/data-types.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+    let numbers: [i32; 5] = [11, 82, 12, 88, 90];
+
+    // Accessing array elements
+    let first_number = numbers[0];
+    let second_number = numbers[1];
+
+    // Printing tuple and array elements
+    println!("Person: {} is {} years old. Adult: {}", name, age, is_adult);
+    println!("Array: [{}, {}]", first_number, second_number);
+}
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:128]:
```suggestion
fn main() {
    // Array
    let numbers: [i32; 5] = [11, 82, 12, 88, 90];

    // Accessing array elements
    let first_number = numbers[0];
    let second_number = numbers[1];

    // Printing tuple and array elements
    println!("Person: {} is {} years old. Adult: {}", name, age, is_adult);
    println!("Array: [{}, {}]", first_number, second_number);
}
```

File: content/rust/concepts/data-types/data-types.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+```
```
Comment (avdhoottt) [inline content/rust/concepts/data-types/data-types.md:129]: Add the output of the code.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @moha230, can you make the suggested changes here?

**Comment by moha230** [comment]
@avdhoottt yes I make the changes as soon as possible thank you .

File: content/rust/concepts/data-types/data-types.md (hunk 5 above)
Comment (moha230) [reply content/rust/concepts/data-types/data-types.md:17]: yes i i was wrong its scalar not scala

**Comment by moha230** [comment]
@avdhoottt its done Thank you

**Review by avdhoottt — APPROVED** [review]
LGTM!!

**Comment by ishg-153** [comment]
@AF7626 Please sign the CLA.
@moha230 if this a spare email or account you have please sign CLA with that account

**Comment by ishg-153** [comment]
@moha230 This PR is being closed due to a lack of activity.

**Pull request closed without merging**